reconcile/test/test_terraform_vpc_resources.py

```diff
@@ -0,0 +1,127 @@
+import logging
+
+import pytest
+
+from reconcile.gql_definitions.common.app_interface_vault_settings import (
+    AppInterfaceSettingsV1,
+)
+from reconcile.gql_definitions.terraform_vpc_resources.vpc_resources_aws_accounts import (
+    AWSAccountV1,
+    AWSTerraformStateIntegrationsV1,
+    TerraformStateAWSV1,
+    VPCResourcesAWSAccountsQueryData,
+)
+from reconcile.status import ExitCodes
+from reconcile.terraform_vpc_resources import (
+    TerraformVpcResources,
+    TerraformVpcResourcesParams,
+)
+
+
+@pytest.fixture
+def query_data():
+    return VPCResourcesAWSAccountsQueryData(
+        accounts=[
+            AWSAccountV1(
+                name="some-account",
+                terraformState=TerraformStateAWSV1(
+                    integrations=[
+                        AWSTerraformStateIntegrationsV1(
+                            key="some-key",
+                            integration="some-integration",
+                        ),
+                    ]
+                ),
+            ),
+            AWSAccountV1(
+                name="some-other-account",
+                terraformState=TerraformStateAWSV1(
+                    integrations=[
+                        AWSTerraformStateIntegrationsV1(
+                            key="terraform-vpc-resouces.tf.json",
+                            integration="terraform-vpc-resources",
+                        ),
+                    ]
+                ),
+            ),
+        ]
+    )
+
+
+@pytest.fixture
+def mock_gql(mocker):
+    return mocker.patch(
+        "reconcile.terraform_vpc_resources.gql",
+        autospec=True,
+    )
+
+
+@pytest.fixture
+def mock_app_interface_vault_settings(mocker):
+    mocked_app_interfafe_vault_settings = mocker.patch(
+        "reconcile.terraform_vpc_resources.get_app_interface_vault_settings",
+        autospec=True,
+    )
+    mocked_app_interfafe_vault_settings.return_value = AppInterfaceSettingsV1(
+        vault=True
+    )
+    return mocked_app_interfafe_vault_settings
+
+
+@pytest.fixture
+def mock_create_secret_reader(mocker):
+    return mocker.patch(
+        "reconcile.terraform_vpc_resources.create_secret_reader", autospec=True
+    )
```

Comment:
nit: Personal recommendation is to use dependency injection pattern instead of mock patching. I.e., 

```python
from unittest.mock import create_autospec

mocked_secret_reader = create_autospec(spec=SecretReaderBase)
integration = Integration(secret_reader=mocked_secret_reader)

mocked_secret_reader.read_secret.assert_called_once_with(...)
```

This gives you a clean mock object that will throw errors if it is used in a non-supported way. You can find this pattern in many integrations. That being said, we must resort to mock patching for older components that were not built with dependency injection in mind.

Reply:
I like dependency injection but I not finding any integration that uses this in the integration constructor. I see many examples of the secret reader being created on the `run` method and being passed as parameter to other methods.

Reply:
At least SAPM uses that pattern :)

Reply:
I'm going to take a look into that.